### PR TITLE
Part of 17670: Replace Typography with Text in detected-token-address.js

### PR DESCRIPTION
**Pull request by Ananyamadhu08**
## Explanation

Updating Typography to Text `detected-token-address.js`

Part of: #17670

### Before

Screenshot 2023-04-04 at 4 10 25 AM

### After

Screenshot 2023-04-04 at 4 09 02 AM

## Pre-merge author checklist

- [x] I've clearly explained:
  - [x] What problem this PR is solving
  - [x] How this problem was solved
  - [ ] How reviewers can test my changes
- [ ] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [x] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [x] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

**Comment by Ananyamadhu08** [comment]
I have read the CLA Document and I hereby sign the CLA

File: ui/components/app/detected-token/detected-token-address/detected-token-address.js [hunk 1]
```diff
@@ -6,34 +6,35 @@ import { useCopyToClipboard } from '../../../../hooks/useCopyToClipboard';
...
+import { Text } from '../../../component-library';
 
 const DetectedTokenAddress = ({ tokenAddress }) => {
   const t = useI18nContext();
   const [copied, handleCopy] = useCopyToClipboard();
 
   return (
     <Box display={DISPLAY.INLINE_FLEX} className="detected-token-address">
-      <Typography variant={TypographyVariant.H7} color={TextColor.textDefault}>
+      <Text variant={TextVariant.bodySm} as="h7" color={TextColor.textDefault}>
```
Comment (georgewrmarshall) [inline ui/components/app/detected-token/detected-token-address/detected-token-address.js:26]: Update to `h6` tag
```suggestion
      <Text variant={TextVariant.bodySm} as="h6" color={TextColor.textDefault}>
```

File: ui/components/app/detected-token/detected-token-address/detected-token-address.js [hunk 2]
```diff
@@ -6,34 +6,35 @@ import { useCopyToClipboard } from '../../../../hooks/useCopyToClipboard';
...
         {`${t('tokenAddress')}:`}
-      </Typography>
-      <Typography
-        variant={TypographyVariant.H7}
+      </Text>
+      <Text
+        variant={TextVariant.bodySm}
+        as="h7"
```
Comment (georgewrmarshall) [inline ui/components/app/detected-token/detected-token-address/detected-token-address.js:31]: Update to `h6` tag
```suggestion
        as="h6"
```

**Review by georgewrmarshall — CHANGES_REQUESTED** [review]
Thanks for your contribution! Excellent PR description and screenshots 💯 There was a mistake in the issue that incorrectly mapped the `Text` component to an `h7` tag. Heading tags in html go from `h1` - `h6`. I've update the issue and made a couple of suggestions for the tags. Apologies! If you can make those 2 changes I can approve 👍

**Comment by Ananyamadhu08** [comment]
@georgewrmarshall I have made the requested changes, please review.

**Review by georgewrmarshall — APPROVED** [review]
LGTM 💯 Thanks for your contribution!
- checked storybook on local

**Review by NidhiKJha — APPROVED** [review]
LGTM

**Pull request merged**